src/compiler/checker.ts

```diff
@@ -28959,6 +28959,15 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                     return isCircularMappedProperty(prop) ? undefined : getTypeOfSymbol(prop);
                 }
                 if (isTupleType(t) && isNumericLiteralName(name) && +name >= 0) {
+                    if (t.target.hasRestElement) {

```

Comment:
The tuple type could be `[...T]` where `T extends []` and, thus, should _not_ have a contextual type for a numbered element (at all), so I'm not sure checking this field is correct.